zeppelin-integration/src/test/java/org/apache/zeppelin/ZeppelinITUtils.java

```diff
@@ -58,4 +69,58 @@ public static void turnOnImplicitWaits(WebDriver driver) {
     driver.manage().timeouts()
       .implicitlyWait(Duration.ofSeconds(AbstractZeppelinIT.MAX_IMPLICIT_WAIT));
   }
+
+  public static WebElement pollingWait(final WebDriver driver, final By locator, final long timeWait) {
+    Wait<WebDriver> wait = new FluentWait<>(driver)
+        .withTimeout(Duration.of(timeWait, ChronoUnit.SECONDS))
+        .pollingEvery(Duration.of(1, ChronoUnit.SECONDS))
+        .ignoring(NoSuchElementException.class);
+
+    return wait.until(new Function<WebDriver, WebElement>() {
+      @Override
+      public WebElement apply(WebDriver driver) {
+        return driver.findElement(locator);
+      }
+    });
+  }
+
+  public static void authenticationUser(WebDriver driver, String userName, String password) {
+    ZeppelinITUtils.pollingWait(
+        driver,
+        By.xpath("//div[contains(@class, 'navbar-collapse')]//li//button[contains(.,'Login')]"),
+        MAX_BROWSER_TIMEOUT_SEC).click();
+
+    ZeppelinITUtils.sleep(1000, false);
+
+    ZeppelinITUtils.pollingWait(
+        driver,
+        By.xpath("//*[@id='userName']"),
+        MAX_BROWSER_TIMEOUT_SEC).sendKeys(userName);
+    ZeppelinITUtils.pollingWait(
+        driver,
+        By.xpath("//*[@id='password']"),
+        MAX_BROWSER_TIMEOUT_SEC).sendKeys(password);
+    ZeppelinITUtils.pollingWait(
+        driver,
+        By.xpath("//*[@id='loginModalContent']//button[contains(.,'Login')]"),
+        MAX_BROWSER_TIMEOUT_SEC).click();
+
+    ZeppelinITUtils.sleep(1000, false);
+  }
+
+  public static void logoutUser(WebDriver driver, String userName) throws URISyntaxException {
```

Comment:
You could also move this function to `AbstractZeppelinIT`.

Reply:
adopted